Release 4.1.0

golluroSICKAG opened the pull request:
# Release 4.1.0

## Improvements
- Deactivate signal links for SIM300 (not supported)

## Bugfix
- Legacy bindings of ValueDisplay elements within UI did not work if deployed with VS Code AppSpace SDK
- UI differs if deployed via Appstudio or VS Code AppSpace SDK
- Fullscreen icon of iFrame was visible

## Following tasks were checked

- [x] Module is coded and structured according to the "Developing guideline for modules" described within the CSK documentation
- [x] All functions/events/parameters are documented within the manifest documentation
- [x] The manifest description of the main CROWN includes main information about the purpose of the module and how to use it in general
- [x] API docu based on the manifest was created and stored within the "docu"-folder of the repository
- [x] Internal LUA code documentation exists for variables and non served functions
- [x] All relevant infos are logged via the SharedLogger 'ModuleLogger'
- [x] Module supports persistent data feature based on 'CSK_Module_PersistentData'
- [x] Module supports user management based on 'CSK_Module_UserManagement'
- [x] No open "ToDos" within the code or at least clearly explained comments why they exist ...
- [x] "Version" key in app manifest was updated following semantic versioning (and use '0.x.y' for test / experimental modules which are not yet ready to be officially released)
- [x] Meaningful IDs used for UI elements
- [x] Module was tested on an AppSpace device (at least on SICK AppEngine) with no error message
- [x] README.md is up to date (incl. info of device + firmware the module was tested with)
- [x] CHANGELOG.md is up to date

golluroSICKAG committed:
# Release 4.0.0

## New features
- Supports FlowConfig to link digital IOs
- Provide version of module via 'OnNewStatusModuleVersion'
- Function 'getParameters' to provide PersistentData parameters
- Check if features of module can be used on device and provide this via 'OnNewStatusModuleIsActive' event / 'getStatusModuleActive' function
- Added functions 'removeForwardEventViaUI' and 'removeTriggerEventViaUI'. Furthermore 'removeTriggerEvent' and 'removeForwardEvent' require now the eventname as first parameter
- Function to 'resetModule' to default setup

## Improvements
- New UI design available (e.g. selectable via CSK_Module_PersistentData v4.1.0 or higher), see 'OnNewStatusCSKStyle'
- 'loadParameters' returns its success
- 'sendParameters' can control if sent data should be saved directly by CSK_Module_PersistentData
- Added UI icon
- Changed log level of some messages from 'info' to 'fine'
- Removed 'setForwardEvent' as it was not used

golluroSICKAG committed:
- Update test versions in README

golluroSICKAG committed:
- Update of UI settings

golluroSICKAG committed:
- Fixed FlowConfig ENUM issue

golluroSICKAG committed:
- Removing navigation bar entries

golluroSICKAG committed:
Merge branch 'SICKAppSpaceCodingStarterKit:main' into main

golluroSICKAG committed:
# Release 4.1.0

## Improvements
- Deactivate signal links for SIM300 (not supported)

## Bugfix
- Legacy bindings of ValueDisplay elements within UI did not work if deployed with VS Code AppSpace SDK
- UI differs if deployed via Appstudio or VS Code AppSpace SDK
- Fullscreen icon of iFrame was visible

kogliinSICKAG reviewed (approved):
Tested the GUI with AppStudio, no obvious bugs found.
Also tested it shortly on SIM300 with inofficial v1.0.0.

golluroSICKAG committed:
- Reset function did not clear all signal links
- 'OnNewInputState' block of FlowConfig did not work with other blocks if it was used in parallel with the 'Set' block
